### PR TITLE
[CI] Generate concepts summary

### PR DESCRIPTION
**Pull request by ErikSchierboom**
FYI @mikedamay

### CODE REVIEW

File: .github/bin/concepts/update.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# Checkout to new branch
+BRANCH="conceptssummary-$(date +%Y%m%d%H%M%S)"
+git checkout -b "$BRANCH"
+
+# Setup the git user (required to commit anything)
+git config --global user.email "github-actions[bot]@users.noreply.github.com"
+git config --global user.name "github-actions[bot]"
+
+# Commit and push the changes
+git commit -m "[CI] Update concepts summary"
```
Comment (SaschaMann) [inline .github/bin/concepts/update.sh:28]: Unrelated to this PR: Could we add a `[BOT]` tag or sth like that to easily filter auto-generated commits? `[CI]` is often used for changes to CI.

File: .github/bin/concepts/update.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+git add reference/references.json
+
+# Check if there is nothing to commit (i.e. no changes to the concepts summary)
+if [ -z "$(git status --porcelain)" ]; then
+    echo "No changes to the concepts summary"
+    exit 0
+fi
+
+# Checkout to new branch
+BRANCH="conceptssummary-$(date +%Y%m%d%H%M%S)"
```
Comment (SaschaMann) [inline .github/bin/concepts/update.sh:20]: What do you think about renaming these to `summaries/concept/$date` or similar? That way they don't interfere as much with tab completion.

**Comment by iHiD** [comment]
This is great! :)

File: .github/bin/concepts/update.sh (hunk 2 above)
Comment (ErikSchierboom) [reply .github/bin/concepts/update.sh:20]: I like it. I'll update in a follow-up PR.

File: .github/bin/concepts/update.sh (hunk 1 above)
Comment (ErikSchierboom) [reply .github/bin/concepts/update.sh:28]: Good idea. I'll update in a follow-up PR.

**Pull request merged**